### PR TITLE
[example] Ti-example: simulation of snow growth by phase field method

### PR DESCRIPTION
**Pull request by mo-hanxuan**
Add a file for ti-example: simulation of snow growth by phase field method.

### CODE REVIEW

File: python/taichi/examples/simulation/snow_phaseField.py [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+
+        if self.writeImages:
+            path = "./results/"
+            if not os.path.exists(path): os.makedirs(path)
+
+        for i in range(steps):
+            if i % self.showFrameFrequency == 0:
+                gui_phi.set_image(self.phi)
+                gui_phi.show(path +
+                             "time_{:.4f}s.png".format(i * self.dt) if i %
```
Comment (erizmr) [inline python/taichi/examples/simulation/snow_phaseField.py:211]: Could you please help make it a f-string like `f"time_{i * self.dt:.4f}s.png"` as the pre-commit requires?

Comment (mo-hanxuan) [reply python/taichi/examples/simulation/snow_phaseField.py]: yeah, I will change it

**Comment by erizmr** [comment]
Hi @mo-hanxuan , could you please sign the [Contributor License Agreement](https://cla-assistant.io/taichi-dev/taichi?pullRequest=6284)?

**Comment by mo-hanxuan** [comment]
> Hi @mo-hanxuan , could you please sign the [Contributor License Agreement](https://cla-assistant.io/taichi-dev/taichi?pullRequest=6284)?

**Comment by mo-hanxuan** [comment]
> Hi @mo-hanxuan , could you please sign the [Contributor License Agreement](https://cla-assistant.io/taichi-dev/taichi?pullRequest=6284)?

Yes, no problem. I just click the above link and signed it. I think this [Contributor License Agreement](https://cla-assistant.io/taichi-dev/taichi?pullRequest=6284)? has been signed now.

**Review by erizmr — APPROVED** [review]
Thanks!

**Comment by feisuzhu** [comment]
@mo-hanxuan

Could you please manually squash your commits, or correct the author of `Ti-example, phase field simulation of snow growth. Format has been changed according to pre-commit` ? Merging is now blocked by CLA assistant because of the incorrect author field.

**Comment by mo-hanxuan** [comment]
> @mo-hanxuan
> 
> Could you please manually squash your commits, or correct the author of `Ti-example, phase field simulation of snow growth. Format has been changed according to pre-commit` ? Merging is now blocked by CLA assistant because of the incorrect author field.

yes, I will try both methods.

**Comment by feisuzhu** [comment]
Yikes, things gone wild ;)

Try following these steps if you have no idea what goes wrong:

1. Backup your current branch, `$ git checkout -b backup`
2. Switch back to your feature branch, in this case `master` (please don't commit directly to `master` next time)
3. Reset your branch to where it is still normal, according to github history, should be `4d4f3eae3a244ba52eb5490350871d6bb27f6ae8` (`$ git reset --hard 4d4f3eae3a244ba52eb5490350871d6bb27f6ae8`)
4. Fetch upstream/master, upstream refers to `https://github.com/taichi-dev/taichi`
5. `git rebase upstream/master`
6. `git rebase -i upstream/master` and change the offending commit from `pick` to `edit`
7. `git commit --amend --author="mohanxuan <1175448684@qq.com>" --no-edit
8. `git rebase --continue`
9. `git push origin +master` # this is your repo and branch

**Comment by mo-hanxuan** [comment]
> @mo-hanxuan
> 
> Could you please manually squash your commits, or correct the author of `Ti-example, phase field simulation of snow growth. Format has been changed according to pre-commit` ? Merging is now blocked by CLA assistant because of the incorrect author field.

It seems that both methods are very difficult to tackle. I am trying to make another PR instead, which seems to be easier

**Comment by mo-hanxuan** [comment]
> Yikes, things gone wild ;)
> 
> Try following these steps if you have no idea what goes wrong:
> 
> 1. Backup your current branch, `$ git checkout -b backup`
> 2. Switch back to your feature branch, in this case `master` (please don't commit directly to `master` next time)
> 3. Reset your branch to where it is still normal, according to github history, should be `4d4f3eae3a244ba52eb5490350871d6bb27f6ae8` (`$ git reset --hard 4d4f3eae3a244ba52eb5490350871d6bb27f6ae8`)
> 4. Fetch upstream/master, upstream refers to `https://github.com/taichi-dev/taichi`
> 5. `git rebase upstream/master`
> 6. `git rebase -i upstream/master` and change the offending commit from `pick` to `edit`
> 7. `git commit --amend --author="mohanxuan [1175448684@qq.com](mailto:1175448684@qq.com)" --no-edit
> 8. `git rebase --continue`
> 9. `git push origin +master` # this is your repo and branch

OK, I will try to follow this step by step

**Comment by mo-hanxuan** [comment]
> > @mo-hanxuan
> > Could you please manually squash your commits, or correct the author of `Ti-example, phase field simulation of snow growth. Format has been changed according to pre-commit` ? Merging is now blocked by CLA assistant because of the incorrect author field.
> 
> yes, I will try both methods.

I replace this by a new PR, which seems easier to solve this problem of CLA signing.

**Pull request closed without merging**